app/code/Magento/Msrp/view/base/web/js/msrp.js

```diff
@@ -266,9 +288,108 @@ define([
             if (this.options.addToCartUrl) {
                 $('.mage-dropdown-dialog > .ui-dialog-content').dropdownDialog('close');
             }
+
+            e.preventDefault();
             $(this.options.cartForm).submit();
+        },
+
+        /**
+         * Call on event updatePrice. Proxy to updateMsrpPrice method.
+         *
+         * @param {Event} event
+         * @param {mixed} priceIndex
+         * @param {Object} prices
+         */
+        onUpdateMsrpPrice: function onUpdateMsrpPrice(event, priceIndex, prices) {
+
+            var defaultMsrp,
+                defaultPrice,
+                msrpPrice,
+                finalPrice;
+
+            defaultMsrp = _.chain(prices).map(function (price) {
+                return price.msrpPrice.amount;
+            }).reject(function (p) {
+                return p === null;
+            }).max().value();
+
+            defaultPrice = _.chain(prices).map(function (p) {
+                return p.finalPrice.amount;
+            }).min().value();
+
+            if (typeof prices[priceIndex] !== 'undefined') {
+                msrpPrice = prices[priceIndex].msrpPrice.amount;
+                finalPrice = prices[priceIndex].finalPrice.amount;
+
+                if (msrpPrice === null || msrpPrice <= finalPrice) {
+                    this.updateNonMsrpPrice(priceUtils.formatPrice(finalPrice));
+                } else {
+                    this.updateMsrpPrice(
+                        priceUtils.formatPrice(finalPrice),
+                        priceUtils.formatPrice(msrpPrice),
+                        false);
+                }
+            } else {
+                this.updateMsrpPrice(
+                    priceUtils.formatPrice(defaultPrice),
+                    priceUtils.formatPrice(defaultMsrp),
+                    true);
+            }
+        },
+
+        /**
+         * Update prices for configurable product with MSRP enabled
+         *
+         * @param {String} finalPrice
+         * @param {String} msrpPrice
+         * @param {Boolean} useDefaultPrice
+         */
+        updateMsrpPrice: function (finalPrice, msrpPrice, useDefaultPrice) {
+            var options = this.tierOptions || this.options;
+
+            $(this.options.fallbackPriceContainer).hide();
+            $(this.options.displayPriceContainer).show();
+            $(this.options.mapInfoLinks).show();
+
+            if (useDefaultPrice || !this.wasOpened) {
```

Comment:
Why do you need to know it was opened or not? Also, you can take this information in real-time instead of having additional state